### DISCUSSION TITLE
大佬们可以新增uid 查询武器的功能吗

### DISCUSSION THREAD

**Discussion by paiky** [Arondight/Adachi-BOT#388] [Ideas]
我看到现在有些工具（比如小程序看到）可以查uid价值，里面给出了角色和武器，武器能不能查到呢

**Comment by mark9804** [comment]
`米游社`可以查到当前你装备在角色身上的武器呀

**Comment by paiky** [comment]
我意思是能不能搞个像uid查角色图鉴那样的 查武器图鉴呢

**Comment by Arondight** [comment]
查不到，没有接口，只能看某个人物拿了什么武器

**Comment by paiky** [comment]
哦这样 我去了解下我刚看到的小程序查价值的是什么东东

**Reply by Arondight** [reply]
我猜是拉的你半年的抽卡记录吧

**Reply by paiky** [reply]
> 我猜是拉的你半年的抽卡记录吧

如果是这样 拿个很久没玩过的号测试一下就清楚了

**Comment by mark9804** [comment]
在这种情况下，你可以查到武器和/或圣遗物：
* 武器/圣遗物被装备在人物身上

在这种情况下，你查不到武器和/或圣遗物：
* 武器/圣遗物没有被任何人装备

所以最多只能提供武器等级和精炼等级两个信息，信息量太少了

**Comment by paiky** [comment]
哦 有可能哦 小程序是叫“派蒙百宝箱” 测不了显示网络问题 估计哪里炸了

**Comment by mark9804** [comment]
> 我猜是拉的你半年的抽卡记录吧

那个小程序的用户输入只有一个uid，查不到抽卡记录的。真能只靠UID查到抽卡记录的话米哈游后端今年年终奖倒扣

**Reply by paiky** [reply]
> > 我猜是拉的你半年的抽卡记录吧
> 
> 那个小程序的用户输入只有一个uid，查不到抽卡记录的。真能只靠UID查到抽卡记录的话米哈游后端今年年终奖倒扣

哈哈哈 我也只看到有一个人进去发了图 不知真假 可能是假的

**Reply by paiky** [reply]
里面还有全站排名和全服超过率。。不过用BOT查了下uid倒是真的

**Comment by paiky** [comment]
关于我上面提到的小程序，今天可以查到了，确实是只需要一个uid就能查出具体的包含的五星卡，武器，以及这一期的深渊情况，从而按某个比例估算价值。

**Reply by Arondight** [reply]
你留意一下，可以查到你在背包里，但是没有被角色拿着的武器么，如果只查角色装备的武器，这个项目也能查

**Comment by paiky** [comment]
小程序《派蒙百宝箱》 公众号《提瓦特小助手》

**Comment by paiky** [comment]
我卸下两把五星武器，再查询两次，还是有查询到，不知道和缓存有没有关系。不过，我觉得应该是没关系的，排行榜上前面的氪佬武器一堆，肯定没有把把装上。

**Reply by Arondight** [reply]
这公众号真恶心，还得看视频，他的武器是个假的，查不到角色不装备的武器

**Comment by paiky** [comment]
这样吗，查不到的？？那我再验证一下。小程序的话都这样的，现在很多都恰广告了，不过这个还得全播完就确实是。

**Comment by mark9804** [comment]
如果武器在背包里的话查不到，除非他能登陆你的游戏账号拿到数据。但是这种情况是不可能发生的。

所以我猜他应该和我们一样，用的米游社API。就是只能查到角色装备的武器。

这个小程序就图一乐就好了，恰点流量的红利，算法极其不靠谱，比如我一氪佬朋友深渊猴了七百多把，被它当作练度极低的痛苦号处理了

**Comment by paiky** [comment]
刚验证了下，好像确实是只能查到携带的武器。
能查询出其他武器是不是因为能查询角色带上某一类型武器，进而查询到该武器数量或者详细信息（精炼度）？
账号价值以及深渊什么的不关注。

**Reply by mark9804** [reply]
武器精炼信息我们是有提供的，虽然我经常误认为是武器星级 ~~（不是我写的，不是我的锅）~~
![image](https://user-images.githubusercontent.com/9006264/141884328-4e949c3d-b8b1-41cc-ae91-54526875fd80.png)
武器名称旁边那个几星就代表精几

查到角色身上的武器信息，然后做一个简单的统计，就知道有多少把五星限定武器和五星常驻武器了

**Comment by mark9804** [comment]
这小程序越来越离谱了，怎么抓出来的
![image](https://user-images.githubusercontent.com/9006264/143664589-0d17600e-6899-4380-b728-7b67c29aec6a.jpeg)

**Reply by ID114514** [reply]
> 这小程序越来越离谱了，怎么抓出来的 ![image](https://user-images.githubusercontent.com/9006264/143664589-0d17600e-6899-4380-b728-7b67c29aec6a.jpeg)

手抓的。。。从游戏内搜UID查的，没挂角色在上面就查不到。。。

**Reply by sunist-c** [reply]
挂个Fiddle或者Burp，抓抓米哈游的网络包，模拟原神客户端发包估计能搞出来。~~但是我好像记得这个提瓦特小助手吃相太难看了~~

**Reply by mark9804** [reply]
只要能解决以下问题就可以模拟客户端发包了

1. 鉴权怎么做
2. 数据包怎么初始化
3. 包的数据结构
4. 哪里应该加密
5. 加密算法的逆向工程
6. 哪里需要签名
7. 服务器返回的UDP包如何解析

能整出这个的不是在米哈游干活就是在橘子蹲着了

**Reply by mark9804** [reply]
提瓦特小助手确定用的是安卓大规模云控 + OCR 的方案，理由是 `阿莫斯芝弓`
